### ISSUE TITLE
Improve accessibility of Switch

### ISSUE THREAD

**Issue by stephanschroeter** [porsche-design-system/porsche-design-system#1638]
Description  
---
Label is not a11y compliant in switch component. 
_"We are currently facing an issue when it comes to the p-switch component.
The a11y error that we are receiving is : 'Ensures buttons have discernible text'
The error is coming from here (see screenshot) . In the span  
you can see that there is no text."_
![image](https://user-images.githubusercontent.com/42934679/146007680-5443769f-cd67-4ac5-8c54-2b04f8be902c.png)
https://porschedev.slack.com/archives/CT7AVHTTQ/p1634744299158600

To Reproduce
---
This issue can only be reproduced by checking it with AXE DevTools. 

This is regarding to the following issue in AXE Core on Github: https://github.com/dequelabs/axe-core/issues/2488
The Dragon Software does not support `button` `label` pairs like this:
``` 
<button role="switch" aria-checked="true" id="switch"></button>
<label for="switch">Speaker power</label>
```

**But:** If checking the accessibility tree, the button gets the correct name. 

**Chrome:**
<img width="953" alt="Bildschirmfoto 2021-12-16 um 08 56 20" src="https://user-images.githubusercontent.com/40788023/146330916-cb395f81-ec39-45bc-b7cc-3e1aff098d52.png">

**Firefox:**
<img width="1330" alt="Bildschirmfoto 2021-12-16 um 08 55 03" src="https://user-images.githubusercontent.com/40788023/146330919-c8bb04df-4ee8-4147-a3b2-d96b95d2edc2.png">

Expected behavior
---

Screenshots
---

Test Environment
---
Platform/Browser/Version: [e.g. iOS Safari 11, macOS Chrome 81]
